src/main/java/com/google/devtools/build/lib/bazel/repository/ProxyHelper.java

```diff
@@ -0,0 +1,129 @@
+// Copyright 2014 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.bazel.repository;
+
+import com.google.common.base.Strings;
+import java.io.IOException;
+import java.net.Authenticator;
+import java.net.InetSocketAddress;
+import java.net.PasswordAuthentication;
+import java.net.Proxy;
+import java.net.URL;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * Helper class for setting up a proxy server for network communication
+ *
+ */
+public class ProxyHelper {
+
+  /**
+   * This method takes a java.net.URL object for the resource being requested and sets up a proxy to make
+   * the request if HTTP_PROXY and/or HTTPS_PROXY environment variables are set.
+   * @param requestedUrl The URL for the remote resource that may need to be retrieved through a proxy
+   * @return Proxy
+   * @throws IOException
+   */
+  public static Proxy createProxyIfNeeded(URL requestedUrl) throws IOException {
```

Comment:
This one is going to be a problem due to the usage of environment variables.  I originally had one for this method but threw my hands up after noticing the environment variables aren't set in Junit execution.  I could wrap the environment variables into another class that can be mocked but then that class assumes the problems with testing in Junit.  Do you have another suggestion on that?